modules/basics/src/main/java/com/opengamma/basics/date/HolidayCalendar.java

```diff
@@ -202,6 +203,30 @@ public default LocalDate previousOrSame(LocalDate date) {
 
   //-------------------------------------------------------------------------
   /**
+   * Finds the next business day within the month, returning the input date if it is a business day,
+   * or the last business day of the month if the next business day is in a different month.
+   * <p>
+   * Given a date, this method returns a business day.
+   * If the input date is a business day, it is returned.
+   * If the next business day is within the same month, it is returned.
+   * Otherwise, the last business day of the month is returned.
+   * <p>
+   * This corresponds to the {@linkplain BusinessDayConventions#MODIFIED_FOLLOWING modified following}
+   * business day convention.
+   * 
+   * @param date  the date to adjust
+   * @return the input date if it is a business day, the next business day if within the same month
+   * or the last business day of the month
+   * @throws IllegalArgumentException if the calculation is outside the supported range
+   */
+  public default LocalDate nextLastOrSameInMonth(LocalDate date) {
```

Comment:
Might ``nextSameOrLastInMonth`` be a better name, given this is how it is documented and implemented?